fetchmap/index.ts

```diff
@@ -0,0 +1,265 @@
+    const existingLegend = document.querySelector('.legend-wrapper');
+    if (existingLegend) {
+      existingLegend.remove();
+    }
+    deck.setProps({layers: []});
+    currentMapData = null;
+
+    if (selectedMapIds.length === 0) {
+      if (mapNameEl) mapNameEl.innerHTML = 'No Map Selected';
+      if (layerCountEl) layerCountEl.innerHTML = '0';
+      map.jumpTo({
+        center: [INITIAL_VIEW_STATE.longitude, INITIAL_VIEW_STATE.latitude],
+        zoom: INITIAL_VIEW_STATE.zoom,
+        bearing: INITIAL_VIEW_STATE.bearing,
+        pitch: INITIAL_VIEW_STATE.pitch
+      });
+      deck.setProps({initialViewState: INITIAL_VIEW_STATE});
+      return;
+    }
+
+    // Fetch map data for each selected map
+    const allFetchedMapData: FetchMapResult[] = [];
+    for (const mapId of selectedMapIds) {
+      try {
+        const fetchOptions = {...baseFetchMapOptions, cartoMapId: mapId};
+        const mapData = await fetchMap(fetchOptions);
+        allFetchedMapData.push(mapData);
+      } catch (error) {
+        // Optionally, show a partial error or skip this map
+      }
+    }
+
+    // If no maps were fetched, show an error
+    if (allFetchedMapData.length === 0) {
+      if (mapNameEl) mapNameEl.innerHTML = 'Error loading maps';
+      if (layerCountEl) layerCountEl.innerHTML = 'Error';
+      return;
+    }
+
+    // Aggregate data from all fetched maps
+    const aggregatedLayers: LayerDescriptor[] = [];
+    const mergedPopupSettingsLayers: NonNullable<
+      NonNullable<FetchMapResult['popupSettings']>['layers']
+    > = {};
+    let finalInitialViewState = {...INITIAL_VIEW_STATE};
+    const mapTitles: string[] = [];
+
+    allFetchedMapData.forEach((mapData, index) => {
+      mapTitles.push(mapData.title || `Map ${index + 1}`);
+      if (mapData.layers) {
+        aggregatedLayers.push(...mapData.layers);
+      }
+      if (mapData.popupSettings && mapData.popupSettings.layers) {
+        for (const layerId in mapData.popupSettings.layers) {
+          // Simple merge: last one wins if layer IDs conflict across maps.
+          mergedPopupSettingsLayers[layerId] = mapData.popupSettings.layers[layerId];
+        }
+      }
+
+      // Use initial view state from the first successfully fetched map
+      if (index === 0) {
+        // Prioritize the first map's view state
+        if (mapData.initialViewState) {
+          finalInitialViewState = {...INITIAL_VIEW_STATE, ...mapData.initialViewState};
+        } else if ((mapData as any).mapOptions && (mapData as any).mapOptions.viewState) {
+          finalInitialViewState = {
+            longitude: (mapData as any).mapOptions.viewState.longitude,
+            latitude: (mapData as any).mapOptions.viewState.latitude,
+            zoom: (mapData as any).mapOptions.viewState.zoom,
+            pitch: (mapData as any).mapOptions.viewState.pitch || 0,
+            bearing: (mapData as any).mapOptions.viewState.bearing || 0
+          };
+        }
+      }
```

Comment:
Wondering if we need the `else if` case here?  I guess I'm not sure what the difference is. 😕 

Possibly the `index === 0` block could be moved outside the loop as:

```javascript
const finalInitialViewState = {
  ...INITIAL_VIEW_STATE,
  ...allFetchedMapData[0].initialViewState
};
```